src/test/java/com/facebook/genAlgo/mutator/MultipleMutatorTest.java

```diff
@@ -31,6 +31,8 @@ void mutateGuaranteed(int geneValue) {
         Gene gene = new Gene(randomProvider);
         gene.setValue((char) geneValue);
         int initialGeneValue = gene.getValue();
+        when(randomProvider.getInt(anyInt())).thenReturn(1);
+        when(randomProvider.getInt(anyInt())).thenReturn(1);
```

Comment:
First getInt() value describes how many times mutation occur so any value other than 0 is needed. 
Another getInt() determines bit position to change. Since this method checks if any change occurs, bit position is irreleveant - even 0 should change value.

Reply:
Yes I agree but random provider is a mock object So without when metod randomProvider do nothing. In this test case we could use normal object.
I also wondering about use spy insted of mock object. I am not sure but spy is a partial mock So i think that spy may know its own metod and act as metod describe. Do you agree? Did you used spy before?
I never used spy :D

Reply:
I believe that randomProvider.getInt() even in mock should do something - return default value, so 0.
You do not need to repeat this line:
when(randomProvider.getInt(anyInt())).thenReturn(1);

It means that every time getInt() on mock will return 1.

Reply:
I once tried to use spy but decided to go with ArgumentCaptor class. But if you have an idea, please go with it :) I am also curious.